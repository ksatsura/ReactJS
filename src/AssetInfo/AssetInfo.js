//imports
import React from 'react';
import '../style.css';

const AssetInfo = (props) => {

    const { asset } = props;

    return (
        <div className='asset-data'>
            <div className='asset-name-rating'>
                <p className='asset-name'>{asset.title}</p>
                <span className='asset-rating'>{asset.vote_average}</span>
            </div>
            <p className='asset-tagline'>{asset.tagline}</p>
            <div className='asset-date-duration'>
                <p className='asset-date'>{asset.release_date.slice(0,4)}</p>
                <p className='asset-duration'>{`${asset.runtime} min`}</p>
            </div>
            <p className='asset-description'>{asset.overview}</p>
        </div>
    );
};

export default AssetInfo;