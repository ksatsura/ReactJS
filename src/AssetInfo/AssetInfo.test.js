import React from 'react';
import AssetInfo from './AssetInfo';
import {shallow} from 'enzyme';

describe('AssetInfo.js', () => {

    const props = {
        asset: {
            title: '',
            release_date: '',
            genre: '',
            vote_average: '',
            tagline: '',
            runtime: '',
            overview: '',
        },
    };

    describe('renders', () => {

        it('renders for desktop', () => {
            const component = shallow(<AssetInfo {...props} />);
            expect(component).toMatchSnapshot();
        });

        it('renders for mobile', () => {
            const component = shallow(<AssetInfo {...props} isMobile />);

            expect(component).toMatchSnapshot();
        });

    });
});
