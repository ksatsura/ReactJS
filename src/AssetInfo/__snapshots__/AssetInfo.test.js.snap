// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetInfo.js renders renders for desktop 1`] = `ShallowWrapper {}`;

exports[`AssetInfo.js renders renders for mobile 1`] = `ShallowWrapper {}`;
